### PR TITLE
Fix moving list item also moves image gallery item

### LINKED ISSUE #36256: Reordering the first Attribute item also moves the cover image
### Describe the bug

In the `Attributes` section, reordering the first list item also moves the cover image.

Screen Capture on 2023-01-02 at 11-31-33

### Steps to reproduce

1. Add a couple of attributes and a cover image to a product.
2. Reorder the attributes, grab the first item and move it.
3. The cover image will move with the draggable item.

### PR DESCRIPTION
**Pull request by octaedro**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

There was a problem with the sortable lists. It wasn't possible to use more than one without strange behavior.
E.g.:
Screen Capture on 2023-01-02 at 11-31-33

That happened because the id used to drag and drop items was the same for both lists.
This PR fixes that problem by adding a unique suffix to the list ids.

Closes #36256.

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Add a couple of attributes and a cover image to a product.
2. Reorder the attributes, grab the first item and move it.
3. The cover image won't move with the draggable list item.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/components/src/sortable/sortable.tsx [hunk 1]
```diff
@@ -265,7 +266,7 @@ export const Sortable = ( {
 						<SortableItem
 							key={ child.key || index }
 							className={ itemClasses }
-							id={ index }
+							id={ `${ index }${ v4() }` }
```
Comment (joshuatf) [inline packages/js/components/src/sortable/sortable.tsx]: Optional, but I think we could remove `index` or add a `-` here if we need to keep the index for debugging.

**Review by joshuatf — DISMISSED** [review]
Nice fix!  Testing well for me.  Left one minor tweak and looks like we'll need to rebase some conflicts, but pre-approving ✅

File: packages/js/components/src/sortable/sortable.tsx (hunk 1 above)
Comment (octaedro) [reply packages/js/components/src/sortable/sortable.tsx]: Sounds good to me! I added the `-` between the item `index` and the `id` in the commit [248e3e1](https://github.com/woocommerce/woocommerce/pull/36262/commits/248e3e1eb07754af622a1575ae5cb614de45f8a9)

**Comment by octaedro** [comment]
Hi @joshuatf, I added the change you mentioned and rebased this PR, so I think that it will need another approval.

**Review by joshuatf — APPROVED** [review]
Thanks for the change!  LGTM 🚢

**Pull request merged**